packages/jira-adapter/src/filters/script_runner/workflow_scripts.ts

```
@@ -0,0 +1,182 @@
+      reject(err)
+    } else {
+      resolve(result)
+    }
+  })
+})
+
+type CompressedObject = {
+  compressed: number[]
+}
+
+const COMPRESSED_OBJECT_SCHEME = Joi.object({
+  compressed: Joi.array().items(Joi.number()).required(),
+})
+
+const isCompressedObject = createSchemeGuard<CompressedObject>(COMPRESSED_OBJECT_SCHEME, 'ScriptRunner object not as expected')
+
+// the script runner decode path
+// stringBase64=>serializedString=>compressedObject=>dataString=>dataObject
+// it starts with a base64 string. Decoding it turns into a json serialized string
+// the json is in format of {"compressed: Array<number>"}. The array is a gzip compressed buffer
+// decompressing the buffer turns it into a json string of the data object
+
+const decodeScriptRunner = async (scriptRunnerString: string | undefined): Promise<Value> => {
+  if (scriptRunnerString === undefined) {
+    return undefined
+  }
+  try {
+    const compressedObject = JSON.parse(Buffer.from(scriptRunnerString, 'base64').toString('utf8'))
+    if (!isCompressedObject(compressedObject)) {
+      return scriptRunnerString
+    }
+    const zipBuffer = Buffer.from(compressedObject.compressed)
+    const dataString = (await unzip(zipBuffer)).toString()
+    return JSON.parse(dataString)
+  } catch (e) {
+    log.warn('Could not decode script runner')
+    if (e instanceof Error) {
+      log.warn('Error due to  %s', e.message)
+    }
+    return scriptRunnerString
+  }
+}
+
+const encodeScriptRunner = async (object: Value): Promise<Value> => {
+  try {
+    const dataString = safeJsonStringify(object)
+    const zipBuffer = await zip(Buffer.from(dataString))
+    const compressedObject = {
+      compressed: zipBuffer.toJSON().data,
+    }
+    return Buffer.from(safeJsonStringify(compressedObject)).toString('base64')
+  } catch (e) {
+    log.warn('Could not encode script runner object')
+    if (e instanceof Error) {
+      log.warn('error due to  %s', e.message)
+    }
+    return object
```

I think in this case we really want to throw an error since deploying it as an object might mess up the workflow